temporian/core/operators/combine.py

```diff
@@ -0,0 +1,236 @@
+from temporian.core.compilation import compile
+from temporian.core.data.node import (
+    EventSetNode,
+    create_node_new_features_new_sampling,
+)
+from temporian.core.operators.base import Operator
+from temporian.core.typing import EventSetOrNode
+from temporian.proto import core_pb2 as pb
+from temporian.utils.rtcheck import rtcheck
+
+MAX_NUM_ARGUMENTS = 30
+FROM_INTERSECT = "intersect"
+FROM_UNION = "union"
+FROM_FIRST = "first"
+
+
+class Combine(Operator):
+    def __init__(self, index_from: str, **inputs: EventSetNode):
+        super().__init__()
+
+        # Note: Support for dictionaries of nodes is required for
+        # serialization.
+
+        if len(inputs) < 2:
+            raise ValueError("At least two arguments should be provided")
+
+        if len(inputs) >= MAX_NUM_ARGUMENTS:
+            raise ValueError(
+                f"Too many (>{MAX_NUM_ARGUMENTS}) arguments provided"
+            )
+
+        # Attributes
+        self._index_from = index_from
+        self.add_attribute("index_from", index_from)
+
+        # inputs
+        first_input = None
+        all_unix_timestamp = True
+        for key, input in inputs.items():
+            # Check that all features are in all nodes
+            if first_input is None:
+                first_input = input
+            first_input.schema.check_compatible_features(
+                input.schema, check_order=False
+            )
+            first_input.schema.check_compatible_index(input.schema)
+
+            # Output is unix if all inputs are
+            all_unix_timestamp &= input.schema.is_unix_timestamp
+            self.add_input(key, input)
+
+        assert first_input is not None  # for static checker
+        self.add_output(
+            "output",
+            create_node_new_features_new_sampling(
+                features=first_input.schema.features,
+                indexes=first_input.indexes,
+                is_unix_timestamp=all_unix_timestamp,
+                creator=self,
+            ),
+        )
+        self.check()
+
+    @property
+    def index_from(self) -> str:
+        return self._index_from
+
+    @classmethod
+    def build_op_definition(cls) -> pb.OperatorDef:
+        return pb.OperatorDef(
+            key="COMBINE",
+            attributes=[
+                pb.OperatorDef.Attribute(
+                    key="index_from",
+                    type=pb.OperatorDef.Attribute.Type.STRING,
+                ),
+            ],
+            inputs=[
+                pb.OperatorDef.Input(key=f"input_{idx}", is_optional=idx >= 2)
+                for idx in range(MAX_NUM_ARGUMENTS)
+            ],
+            outputs=[pb.OperatorDef.Output(key="output")],
+        )
+
+
+operator_lib.register_operator(Combine)
+
+
+@rtcheck
+@compile
+def combine(
+    *inputs: EventSetOrNode,
+    index_from: str = "union",
+) -> EventSetOrNode:
+    """Combines all events from multiple EventSets with the same features.
```

Comment:
Note: The documentation is now in the EventSet's parent class.

Reply:
I think this should be like `glue`, intended to be called as `tp.combine(evset_1, evset_2, evset_3)` (as opposed to `evset_1.combine(evset_2, evset_3)`.
Don't you think so?